blockchains/cardano/cardano_worker.js

```diff
@@ -168,12 +172,11 @@ class CardanoWorker extends BaseWorker {
       // Check if there are new blocks now. We want an interval of at least 2 blocks. For some reason the Cardano Node
       // API would return a partial last block. The same would not happen if the same block is fetched as part of 2
       // block interval.
-      const newConfirmedBlock = await this.getCurrentBlock() - constants.CONFIRMATIONS;
-      if (newConfirmedBlock < fromBlock + 2) {
+      await this.setLastConfirmedBlock();
+      if (this.lastConfirmedBlock < fromBlock + 2) {
         // The Node has not progressed enough
         return [];
       }
-      this.lastConfirmedBlock = newConfirmedBlock;
```

Comment:
No need to go through new variable here, the `this.lastConfirmedBlock` would already be polluted on init() if the Node is broken.